CMakeLists.txt

```diff
@@ -1,74 +1,151 @@
-cmake_minimum_required(VERSION 2.8.3)
-project(bluespace_ai_xsens_mti_driver)
+# Copyright 2023 Haul Vision
 
-cmake_policy(SET CMP0057 NEW)
+cmake_minimum_required(VERSION 3.8)
+project(ros2_xsens_mti_driver)
+
+###############
+## Configure ##
+###############
+
+# Default to C99
+if (NOT CMAKE_C_STANDARD)
+    set(CMAKE_C_STANDARD 99)
+endif ()
 
 # Default to C++14
-if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 14)
-endif()
+if (NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 14)
+endif ()
+
+if (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    add_compile_options(-Wall -Wextra -Wpedantic)
+endif ()
 
 # Find dependencies
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
 find_package(tf2 REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(std_msgs REQUIRED)
 find_package(geometry_msgs REQUIRED)
 find_package(sensor_msgs REQUIRED)
 
+# Add a directory with headers
+include_directories(include)
+include_directories(SYSTEM /opt/ros/$ENV{ROS_DISTRO}/include) # IntelliJ can't see rclcpp without that
```

Comment:
Nice! I don't know why I've never thought of that, I use bash scripts which first source ROS then launch IDE in order to have ROS libs linked 🤦🏼

Reply:
btw. I figured out that when working with docker and Clion, it is enough to follow these two steps:
1. Prepare 'Toolchain' that connects the container over SSH.
2. Gather the following envs:

```shell
ros_env="AMENT_PREFIX_PATH CMAKE_PREFIX_PATH COLCON_PREFIX_PATH PKG_CONFIG_PATH PYTHONPATH LD_LIBRARY_PATH PATH ROS_DISTRO ROS_PYTHON_VERSION ROS_LOCALHOST_ONLY ROS_VERSION"
env_string=""
for e in ${ros_env}; do
env_string+="$e=${!e};"
done
echo "$env_string"
```

and then paste it to the `Environment` field in the `Settings | Build, Execution, Deployment | CMake`.

It works pretty ok for me. Sometimes the IntelliJ gets crazy and can't see `rclcpp`, but the `include_directories(SYSTEM /opt/ros/$ENV{ROS_DISTRO}/include)` (which isn't necessary to properly build the package) should fix it.